# **df5 - Customer Segmentation Dataset (Unsupervised Learning - Clustering)**

# **Description: Used for clustering customers into risk profiles.**

In [2]:
import pandas as pd
df5=pd.read_csv(r'/content/drive/MyDrive/Captsone project/Data/df5-customer_segmentation_dataset.csv')
df5.head()

Unnamed: 0                           Customer_ID  Age     Income  \
0           0  bdd640fb-0667-4ad1-9c80-317fa3b1799d   58   40039.59   
1           1  bd9c66b3-ad3c-4d6d-9a3d-1fa7bc8960a9   61  153320.19   
2           2  815ef6d1-3b8f-4a18-b7f8-a88b17fc695a   19   36865.14   
3           3  96da1dac-72ff-4d2a-b86e-cbe06b65a6a4   63  136979.20   
4           4  b2b9437a-28df-4ec4-8e4a-2bbdc241330b   69  176474.06   

         Location  Number_of_Active_Policies  Total_Premium_Paid  \
0        Johnberg                          3            12999.70   
1  New Roberttown                          5             5260.00   
2   New Jamesside                          2            25762.41   
3        Lisatown                          5            21556.47   
4    Petersonberg                          2            35208.83   

   Claim_Frequency  Policy_Upgrades  
0                2                0  
1                6                0  
2                0                1  
3                7                2  
4                5                2

# **SQL - Create Table**

In [4]:
import sqlite3
import pandas as pd

# File name of the uploaded CSV
csv_filename = "/content/drive/MyDrive/Captsone project/Data/df5-customer_segmentation_dataset.csv"  # Replace this with the uploaded file name

# Load CSV into a pandas DataFrame
df = pd.read_csv(csv_filename)

# Connect to SQLite database (or create a new one)
conn = sqlite3.connect("example.db")
cursor = conn.cursor()

# Write DataFrame to SQLite table
table_name = "df5"  # Specify your table name
df.to_sql(table_name, conn, if_exists="replace", index=False)

print(f"Table '{table_name}' created in SQLite database.")


Table 'df5' created in SQLite database.


# **1) Retrieve All Data**

In [5]:
query="Select * From df5"

result = pd.read_sql_query(query,conn)
result

Unnamed: 0                           Customer_ID  Age     Income  \
0             0  bdd640fb-0667-4ad1-9c80-317fa3b1799d   58   40039.59   
1             1  bd9c66b3-ad3c-4d6d-9a3d-1fa7bc8960a9   61  153320.19   
2             2  815ef6d1-3b8f-4a18-b7f8-a88b17fc695a   19   36865.14   
3             3  96da1dac-72ff-4d2a-b86e-cbe06b65a6a4   63  136979.20   
4             4  b2b9437a-28df-4ec4-8e4a-2bbdc241330b   69  176474.06   
..          ...                                   ...  ...        ...   
995         995  e98965f2-2bbf-4e35-84ac-b57d0b3d6dc9   58  102911.23   
996         996  2dca365a-2504-4d28-9e6a-e4592980f6ff   77  153665.24   
997         997  1e18046d-2e42-4840-a05f-36c849ed856e   53  116459.39   
998         998  75699aef-0320-4c4e-80cb-e188c3f0fac3   22  194530.39   
999         999  e1f8d192-9ba4-45a6-8178-de26ceb557ac   58  116169.17   

            Location  Number_of_Active_Policies  Total_Premium_Paid  \
0           Johnberg                          3            12999.70   
1     New Roberttown                          5             5260.00   
2      New Jamesside                          2            25762.41   
3           Lisatown                          5            21556.47   
4       Petersonberg                          2            35208.83   
..               ...                        ...                 ...   
995       Arianaberg                          5            44900.50   
996  South Kylemouth                          5            15667.84   
997        East Adam                          2            29292.48   
998        Foleyside                          3            42196.22   
999       Youngshire                          2            28978.10   

     Claim_Frequency  Policy_Upgrades  
0                  2                0  
1                  6                0  
2                  0                1  
3                  7                2  
4                  5                2  
..               ...              ...  
995                7                3  
996               10                2  
997                5                2  
998                5                3  
999                6                1  

[1000 rows x 9 columns]

# **2) Find Customers with More than 3 Active Policies**

In [6]:
query="Select Customer_ID,Age,Location,Number_of_Active_Policies From df5 where Number_of_Active_Policies>=3;"

result = pd.read_sql_query(query,conn)
result

Customer_ID  Age           Location  \
0    bdd640fb-0667-4ad1-9c80-317fa3b1799d   58           Johnberg   
1    bd9c66b3-ad3c-4d6d-9a3d-1fa7bc8960a9   61     New Roberttown   
2    96da1dac-72ff-4d2a-b86e-cbe06b65a6a4   63           Lisatown   
3    562b0f79-c374-49ee-b50b-ea63371ecd7b   27     North Noahstad   
4    43b7a3a6-9a8d-4a03-980d-7b71d8f56413   40  Lake Stephenville   
..                                    ...  ...                ...   
613  e83bcc84-a609-4ad0-b114-64aa52f69081   67       Lake Jessica   
614  730bc657-b4cb-4613-9d39-e9c217584f23   71    Christopherfurt   
615  e98965f2-2bbf-4e35-84ac-b57d0b3d6dc9   58         Arianaberg   
616  2dca365a-2504-4d28-9e6a-e4592980f6ff   77    South Kylemouth   
617  75699aef-0320-4c4e-80cb-e188c3f0fac3   22          Foleyside   

     Number_of_Active_Policies  
0                            3  
1                            5  
2                            5  
3                            3  
4                            5  
..                         ...  
613                          3  
614                          3  
615                          5  
616                          5  
617                          3  

[618 rows x 4 columns]

# **Total 618 Users Have More than 3 Policies Out of 1000 Users**

# **3) Find Customers Who Paid More Than $25000 in Total Premium**

In [7]:
query="Select Customer_ID,Age,Location,Number_of_Active_Policies,Total_Premium_Paid From df5 where Total_Premium_Paid>=25000 Order By Total_Premium_Paid ASC;"

result = pd.read_sql_query(query,conn)
result

Customer_ID  Age         Location  \
0    3a6c7da6-57f7-471f-a978-cc731440adec   57        Loganport   
1    5168bbf2-97af-4bde-bc8e-9d782d14e9f4   80        Robintown   
2    aa1d3fc6-5f31-473d-9519-830c40198303   59      North Randy   
3    25074181-8d1f-4540-b4ef-f5453e652603   42         Leehaven   
4    fa8dae42-ba54-4920-a099-d18087623997   58        Port Adam   
..                                    ...  ...              ...   
483  c5e78b23-f461-4f09-985a-befbd1812f77   74       West Bobby   
484  ca9e4a62-fae3-414b-a3b4-42bd46494296   46  North Emilyfurt   
485  1924c644-af52-4666-b239-09f4dbb54645   64       East Kevin   
486  9b33e3a4-2620-49a9-b4fd-33d184f2fd0f   51       West Sarah   
487  560c95ee-638c-454c-876e-2bba7c5308bf   19      Ramseymouth   

     Number_of_Active_Policies  Total_Premium_Paid  
0                            5            25107.80  
1                            4            25161.02  
2                            2            25249.50  
3                            2            25271.10  
4                            5            25285.42  
..                         ...                 ...  
483                          1            49861.42  
484                          4            49865.86  
485                          4            49875.99  
486                          3            49908.02  
487                          1            49995.48  

[488 rows x 5 columns]

# **Almost 490 Users are Paid More than $25000 Are Premium Out of 1000 Users**

# **4) Calculate Average Income by Age Group**

In [8]:
query="Select Age,avg(Income) From df5 Group by Age;"

result = pd.read_sql_query(query,conn)
result

Age    avg(Income)
0    18   68140.772857
1    19  115301.455000
2    20  106333.218125
3    21   86633.578571
4    22  102859.938571
..  ...            ...
58   76  104706.482500
59   77  121370.102000
60   78  117810.505294
61   79  114489.479500
62   80  113407.488462

[63 rows x 2 columns]

# **5) Find Customers Who Have High Claim Frequency (>8)**

In [9]:
query="Select Customer_ID,Age,Claim_Frequency From df5 where Claim_Frequency>=8;"

result = pd.read_sql_query(query,conn)
result

Customer_ID  Age  Claim_Frequency
0    35a240ae-5af3-4553-9ec4-2e0829a3b2e9   28               10
1    61b1cd22-6280-4c45-9043-5a1098ae4334   42                8
2    ae270da7-02f0-4b90-b143-262fdc5c0eed   61               10
3    e037e5ed-b8db-4672-b42d-47cc00d4af59   50                8
4    deda4e16-1b3d-4d5c-a9a1-fa6f81f76d1c   27                8
..                                    ...  ...              ...
277  8ec57d47-0a0c-44e5-bc34-1b4ffbdc532e   77               10
278  74246baa-2dca-4c82-8660-a3e9736d417c   34                8
279  e83bcc84-a609-4ad0-b114-64aa52f69081   67                9
280  730bc657-b4cb-4613-9d39-e9c217584f23   71               10
281  2dca365a-2504-4d28-9e6a-e4592980f6ff   77               10

[282 rows x 3 columns]

# **Above Claim Frequency Above 8 has 288 Cutomers its Very Bad**

# **Highest Claim Frequency Customers are Almost Fraud**

# **6) Rank Customers Based on Policy Upgrades (Highest First)**

In [10]:
query="Select Customer_ID,Age,Policy_Upgrades From df5 Where Policy_Upgrades>0 Order by Policy_Upgrades DESC;"

result = pd.read_sql_query(query,conn)
result

Customer_ID  Age  Policy_Upgrades
0    43b7a3a6-9a8d-4a03-980d-7b71d8f56413   40                3
1    88bd6407-2bcf-4e01-a28d-efe39bf00273   56                3
2    f16287e4-e9c3-49e0-b602-f8ac10f1bc81   76                3
3    89fa6a68-8fb5-427b-beb7-99193f22faf8   65                3
4    00257ad1-eb22-43dd-87c5-421eec24a3c5   33                3
..                                    ...  ...              ...
751  7dcf60fd-6330-48ed-9a25-27952f191516   61                1
752  65e5f13f-7086-4a45-8a8a-bb17309a083a   49                1
753  4b5f174b-6c26-4b56-aa9e-dda7761ad914   18                1
754  82fc9294-f9a6-4522-8a83-5709ba146757   76                1
755  e1f8d192-9ba4-45a6-8178-de26ceb557ac   58                1

[756 rows x 3 columns]

# **Almost 755 Users are Upgraded Policies Out of 1000 Customers**

In [11]:
query="Select Location,sum(Total_Premium_Paid) as Total_Paid From df5 Group by Location Order by Total_Paid DESC;"

result = pd.read_sql_query(query,conn)
result

Location  Total_Paid
0         Matthewfurt   127035.01
1        South Andrea   100386.12
2    Port Christopher    94080.38
3           Silvatown    91841.10
4            Hallfurt    89163.75
..                ...         ...
958   South Nancyview     1605.65
959      South Robert     1553.57
960      New Johnbury     1352.51
961       West Carlos     1173.11
962       Gregoryview     1082.87

[963 rows x 2 columns]